src/styles/main.scss

```diff
@@ -1,3 +1,83 @@
-body {
+@import "./abstracts/vars";
+@import "./abstracts/mixins";
+@import "./abstracts/functions";
+
+*,
+*::before,
+*::after {
   margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+
+html,
+body {
+  overflow-x: hidden;
+  height: 100vh;
+}
+
+body {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background-image: url("../images/mesh-179.png");
+  background-repeat: no-repeat;
+  background-position: 0 0;
+  background-size: cover;
+  background-attachment: fixed;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: calendarWidth($day-size, $day-gap, $calendar-pdg);
+  gap: $day-gap;
+  padding: $calendar-pdg;
+
+  $days:
+  "tue" 101px,
+  "wed" 202px,
+  "thu" 303px,
+  "fri" 404px,
+  "sat" 505px,
+  "sun" 606px;
+
+@each $name, $size in $days {
+  &--start-day-#{$name} {
+    .calendar__day:nth-child(1) {
+      margin-left: $size;
+    }
+  }
+}
+
+  &__day {
+    @include daySize($day-size);
+    background-color: $day-bg-clr;
+    border: 1px solid $day-bdr-clr;
+    font-size: $font-size;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    transition: all 0.5s cubic-bezier(0.17, 0.84, 0.44, 1);
+    cursor: pointer;
+
+    &:hover {
+      background-color: $day-hvr-clr;
+      transform: translateY(-20px);
+    }
+  }
+}
+
+@for $index from 1 through 31 {
+  .calendar__day:nth-child(#{$index})::before {
+    content: "#{$index}";
+  }
+}
+
+@for $j from 28 through 31 {
+  .calendar--month-length-#{$j} {
+    & :nth-child(n+#{$j + 2}) {
+      display: none;
+    }
+  }
 }
```

Comment:
Put those for loops inside of calendar block. You might also need to change some logic here because in your solution month-length is set to 28 by default but in the calendar, 29 days are displayed :/

Reply:
And it displays 29 days :) The first rule of Fight Club? 'If it works don't touch it ';)

Again thanks very much for advices!